controllers/openshift/openshift.go

```diff
@@ -1,336 +0,0 @@
-					o.Spec.Template.Spec.Containers[index].Command = []string{}
-				}
-			}
-			o.Spec.Template.Spec.InitContainers[0].Args = getArgsForRedhatHaRedisInitContainer()
-			o.Spec.Template.Spec.InitContainers[0].Command = []string{}
-		}
-	case *corev1.Secret:
-		if allowedNamespace(cr.ObjectMeta.Namespace, os.Getenv("ARGOCD_CLUSTER_CONFIG_NAMESPACES")) {
-			logv.Info("configuring cluster secret with empty namespaces to allow cluster resources")
-			delete(o.Data, "namespaces")
-		}
-	case *rbacv1.Role:
-		if o.ObjectMeta.Name == cr.Name+"-"+"argocd-application-controller" {
-			logv.Info("configuring policy rule for Application Controller")
-
-			// can move this to somewhere common eventually, maybe init()
-			k8sClient, err := initK8sClient()
-			if err != nil {
-				logv.Error(err, "failed to initialize kube client")
-				return err
-			}
-
-			clusterRole, err := k8sClient.RbacV1().ClusterRoles().Get(context.TODO(), "admin", metav1.GetOptions{})
-			if err != nil {
-				logv.Error(err, "failed to retrieve Cluster Role admin")
-				return err
-			}
-			policyRules := getPolicyRuleForApplicationController()
-			policyRules = append(policyRules, clusterRole.Rules...)
-			o.Rules = policyRules
-		}
-	}
-	return nil
-}
-
-func getPolicyRuleForApplicationController() []rbacv1.PolicyRule {
-	return []rbacv1.PolicyRule{
-		{
-			APIGroups: []string{
-				"*",
-			},
-			Resources: []string{
-				"*",
-			},
-			Verbs: []string{
-				"get",
-				"list",
-				"watch",
-			},
-		},
-	}
-}
-
-// For OpenShift, we use a custom build of Redis provided by Red Hat
-// which requires additional args in comparison to stock redis.
-func getArgsForRedhatRedis() []string {
-	return []string{
-		"redis-server",
-		"--protected-mode",
-		"no",
-	}
-}
-
-// For OpenShift, we use a custom build of haproxy provided by Red Hat
-// which requires a command as opposed to args in stock haproxy.
-func getCommandForRedhatRedisHaProxy() []string {
-	return []string{
-		"haproxy",
-		"-f",
-		"/usr/local/etc/haproxy/haproxy.cfg",
-	}
-}
-
-// For OpenShift, we use a custom build of Redis provided by Red Hat
-// which requires additional args in comparison to stock redis.
-func getArgsForRedhatHaRedisServer() []string {
-	return []string{
-		"redis-server",
-		"/data/conf/redis.conf",
-	}
-}
-
-// For OpenShift, we use a custom build of Redis provided by Red Hat
-// which requires additional args in comparison to stock redis.
-func getArgsForRedhatHaRedisSentinel() []string {
-	return []string{
-		"redis-sentinel",
-		"/data/conf/sentinel.conf",
-	}
-}
-
-// For OpenShift, we use a custom build of Redis provided by Red Hat
-// which requires additional args in comparison to stock redis.
-func getArgsForRedhatHaRedisInitContainer() []string {
-	return []string{
-		"sh",
-		"/readonly-config/init.sh",
-	}
-}
-
-// policyRulesForClusterConfig defines rules for cluster config.
-func policyRulesForClusterConfig() []rbacv1.PolicyRule {
```

Comment:
Does this mean we are not going to support Cluster Config Argo CD instances with Argo CD Operator ?

Reply:
I want my Argo CD instance to manage OpenShift Cluster Resources. By removing this piece, we will be only left with asking users to create custom cluster roles and bindings.